### ISSUE TITLE
VideoHLS in Cloudflare status, downloadProtection is enabled, how to encode HLS to get thumbs

### ISSUE THREAD

**Issue by chenjian9885** [WWBN/AVideo-Encoder#390]
VideoHLS in Cloudflare status, downloadProtection is enabled, how to encode HLS to get thumbs
![iShot2021-06-09 14 11 52](https://user-images.githubusercontent.com/47063171/121302708-ab9eff00-c92c-11eb-82da-cd50774c4d42.png)

**Comment by chenjian9885** [comment]
I am using the cdn of https://www.cloudflare.com/
How do i fill in
![iShot2021-06-10 23 11 16](https://user-images.githubusercontent.com/47063171/121550298-353ff100-ca41-11eb-9262-55c4a346d108.png)

**Comment by DanielnetoDotCom** [comment]
This should be your endpoint, if you mean you are using a proxied Cloudflare (Not recommended) you should keep it blank

**Comment by DanielnetoDotCom** [comment]
The download protection should work even on a CDN

**Comment by chenjian9885** [comment]
Yes, if the encoder also adds cnd encoding, VideoThumbnails cannot be obtained normally

**Comment by DanielnetoDotCom** [comment]
If you disable the CDN all works fine?
does it fail only when you enable it?

**Comment by chenjian9885** [comment]
Yes it is

**Comment by DanielnetoDotCom** [comment]
again I do not recommend you to use the Cloudflare proxy

maybe you need to disable it. but before disable you may find some clue on the streamer log why it is blocking Cloudflare.

if your encoder is up to date it should send a valid user agent that will be accepted by your streamer, even behind proxies. but cloudflare may remove that information (that is why it is not recommended.)

**Comment by DanielnetoDotCom** [comment]
Again, check your streamer logs, may be a clue somewhere

**Comment by chenjian9885** [comment]
Okay thank you!
